### PR TITLE
iTunes search - state management logic(?)

### PR DESCRIPTION
**Pull request by vishnu-wednesday**
Splitting the large PR into two. This part contains all things state or data into the container.

## GIF

PR#1

### CODE REVIEW

File: app/containers/HomeContainer/index.js [hunk 1]
```diff
@@ -171,10 +171,10 @@ HomeContainer.defaultProps = {
 };
 
 const mapStateToProps = createStructuredSelector({
-  homeContainer: selectHomeContainer(),
-  reposData: selectReposData(),
-  reposError: selectReposError(),
-  repoName: selectRepoName()
+  homeContainer: selectHomeContainer,
+  reposData: selectSearchData(),
```
Comment (alichherawalla) [inline app/containers/HomeContainer/index.js]: let's rename the keys properly

File: app/containers/HomeContainer/reducer.js [hunk 2]
```diff
@@ -4,33 +4,42 @@
...
+      draft.searchTerm = action.searchTerm;
+      break;
+
+    case homeContainerTypes.SUCCESS_GET_TRACKS:
+      draft.searchData = action.data;
+      break;
+
+    case homeContainerTypes.FAILURE_GET_TRACKS:
+      // handle error better
+      draft.searchError = get(action.error, 'message', 'something_went_wrong');
```
Comment (alichherawalla) [inline app/containers/HomeContainer/reducer.js]:
```
draft.songsError = get(action.error, 'message', translate('something_went_wrong'));
```

File: app/containers/HomeContainer/selectors.js [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
  * Other specific selectors
  */
 
 /**
  * Default selector used by HomeContainer
  */
+// is this wrong? https://redux.js.org/usage/deriving-data-selectors#createselector-overview. Read the Warning
+// we have DRY breakage here...
+//export const selectHomeContainer = () => createSelector(selectHomeContainerDomain, (substate) => substate);
+export const selectHomeContainer = (state) => state.homeContainer || iTunesServiceInitialState;
```
Comment (alichherawalla) [inline app/containers/HomeContainer/selectors.js:18]: makes sense, please make the change in react-floki as well, so it generates it correctly

File: app/containers/HomeContainer/selectors.js [hunk 4]
```diff
@@ -1,29 +1,29 @@
...
 
-export const selectHomeContainer = () => createSelector(selectHomeContainerDomain, (substate) => substate);
+export const selectSearchData = () =>
+  createSelector(selectHomeContainer, (substate) => get(substate, 'searchData', null));
 
-export const selectReposData = () =>
-  createSelector(selectHomeContainerDomain, (substate) => get(substate, 'reposData', null));
```
Comment (alichherawalla) [inline app/containers/HomeContainer/selectors.js:22]: let's remove the `null` default value.

File: app/containers/HomeContainer/index.js (hunk 1 above)
Comment (alichherawalla) [reply app/containers/HomeContainer/index.js]: mapDispatchToProps will also have to change

Comment (alichherawalla) [reply app/containers/HomeContainer/index.js]: you'll have to change the name of the keys as well as wire the correct values

Comment (alichherawalla) [reply app/containers/HomeContainer/index.js]: and you can remove the portion responsible for rendering components based on the response of the API in this PR

File: app/services/repoApi.js [hunk 5]
```diff
@@ -1,4 +1,9 @@
 import { generateApiClient } from '@utils/apiUtils';
 const repoApi = generateApiClient('github');
+const iTunesService = generateApiClient('itunes');
```
Comment (alichherawalla) [inline app/services/repoApi.js]: let's rename to itunesAPI

File: app/containers/HomeContainer/selectors.js (hunk 3 above)
Comment (vishnu-wednesday) [reply app/containers/HomeContainer/selectors.js:18]: 👍🏼

File: app/services/repoApi.js (hunk 5 above)
Comment (vishnu-wednesday) [reply app/services/repoApi.js]: 👍🏼

File: app/containers/HomeContainer/index.js (hunk 1 above)
Comment (vishnu-wednesday) [reply app/containers/HomeContainer/index.js]: 👍🏼

**Review by alichherawalla — APPROVED** [review]

**Pull request merged**